woocommerce-gateway-stripe.php

```diff
@@ -42,17 +64,12 @@ function woocommerce_gateway_stripe_init() {
 		return;
 	}
 
+	if ( version_compare( WC_VERSION, WC_STRIPE_MIN_WC_VER, '<' ) ) {
+		add_action( 'admin_notices', 'woocommerce_stripe_wc_not_supported' );
+		return;
+	}
```

Comment:
It seems like this might be related to:
- I think it would be nice to keep the [sanity check notices regarding WooCommmerce versions](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/includes/admin/class-wc-stripe-admin-notices.php#L150-L163). Is there any reason to remove them?

Reply:
> I think it would be nice to keep the sanity check notices regarding WooCommmerce versions. Is there any reason to remove them?

@luizreis I think we need to think about hard vs soft deprecations. In this case, we're doing a hard deprecation. The code you linked to above is better for soft deprecations. Maybe your code could be adapted to make a distinction between these two types? If so we can re-use both soft and hard deprecations in the future, where soft is just a notice and hard a complete block.

Reply:
"deprecated version" vs "unsupported" ?

Reply:
I agree such an approach makes sense to completely avoid loading the plugin. 👍 
As for sanity checks, check of `WC_STRIPE_MIN_WC_VER` doesn't make sense anymore and I'd remove it. But check for `WC_STRIPE_FUTURE_MIN_WC_VER` will make sense once we decide to deprecate some other WC version, so I'd keep it.

Reply:
@dwainm why would we want to have soft deprecation? I wouldn't promote the scenario like `This plugin version can work at your site but it's broken at some places so use it at your own risk!` IMO if the user is smart enough to accept the risk, he/she can easily disable that check and keep using the plugin on their own.

Reply:
@vbelolapotkov the case where it is broken in some places false under hard deprecate. I'm more thinking about the warning in the case where we are aiming to stop supporting it. In the soft case, we can warn users 3 or so releases before the hard deprecation.

Reply:
@dwainm can we use `WC_STRIPE_FUTURE_MIN_WC_VER` for this as we did for 4.3.3 [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/includes/admin/class-wc-stripe-admin-notices.php#L158-L162)?

At the moment the messaging assumes deprecation with next release, but we can alter it to show notification for several releases before the deprecation.

Reply:
@vbelolapotkov Thanks, didn't see that we already have that. That's the right one :) So we have a soft deprecation mechanism in place.

Reply:
Keeping the deprecation notice is the main reason I had for not removing that code. I agree that keeping the `WC_STRIPE_MIN_WC_VER` doesn't make sense, as it's being checked before plugin initialization. I've removed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1189/commits/ca93ae2600037bcb9c86142dc4cf60783465e239.